src/Compilers/Server/VBCSCompiler/Connection.cs

```diff
@@ -104,8 +104,16 @@ public async Task<ConnectionData> HandleConnection(bool allowCompilationRequests
                     LogException(e, "Error reading build request.");
                     return new ConnectionData(CompletionReason.CompilationNotStarted);
                 }
-
-                if (IsShutdownRequest(request))
+                
+                if(request.ProtocolVersion != BuildProtocolConstants.ProtocolVersion)
+                {
+                    return await HandleMismatchedVersionRequest(cancellationToken).ConfigureAwait(false);
+                }
+                else if(request.CompilerHash != BuildProtocolConstants.GetCommitHash())
```

Comment:
Probably should compare this with a case insensitive comparer

Reply:
Should use ordinal vs culture comparison here.

---
In reply to: [215807320](https://github.com/dotnet/roslyn/pull/29711#discussion_r215807320) [](ancestors = 215807320)